wazuh/wazuh-kibana-app
# Add switch to toggle the use of custom branding

# Description

This PR adds a new configuration setting to enable or disable the whole branding customization. This makes it easier to switch on or off all the white-labeling configurations without changing or deleting all the custom branding configurations. 
Changing `customization.enabled` configuration to `true` or `false` is enough to deactivate all the branding customizations.

- Add a new setting:
  - `customization.enabled`: Enable or disable the application of others `customization.*` settings.
  
  
| key                                   | description                                                                                              | values             | requirements on change |
| ---                                     | ---                                                                                                            | ---                    | --- |
| `customization.enabled`  | Enable or disable the application of others `customization.*` settings    | `true`/`false`    | reload browser tab |
  
- Add tests related to the setting

Closes https://github.com/wazuh/wazuh-kibana-app/issues/4506

## Screenshot
![localhost_5601_app_wazuh (1)](https://user-images.githubusercontent.com/34042064/199465225-cc01b418-3f08-4d45-8c7f-45ce40b7c38e.png)

# Test

## Manual tests

**Scenario**: Modify the value of `customization.enabled` setting through `Settings/Configuration`
  **When** the user modifies the `custmization.enabled` setting and saves the configuration.
  **Then** the configuration file should be updated and the frontend
  
**Scenario**: Modify the value of `customization.enabled` setting through an API request with a valid value.
  **When** the user does an API request to update the setting `customization.enabled` with a valid value: `true` or `false`
  **Then** the response should be successful
  **And** the configuration file should be updated
  
**Scenario**: Modify the value of `customization.enabled` setting through an API request with an invalid value.
  **When** the user does an API request to update the setting `customization.enabled` with an invalid value.
    **Then** the response should be a bad request
    
**Scenario**: Customization is applied when the `customization.enabled` setting is enabled.
  **Given** the `customization.enabled` setting value is `true` and the `customization.*` settings have custom values
  **When** the user goes to the plugin,
  **Then** the menu displays the custom image
  **And** the plugin category displays the custom image
  **And** the plugin health check displays the custom image
  **And** the PDF report image displays the custom image
  **And** the PDF header displays the custom text
  **And** the PDF footer displays the custom text
  
**Scenario**: Customization is not applied when the `customization.enabled` setting is disabled.
  **Given** the `customization.enabled` setting value is `false` and the `customization.*` settings have custom values
  **When** the user goes to the plugin,
  **Then** the menu displays the default image
  **And** the plugin category displays the default image
  **And** the plugin health check displays the default image
  **And** the PDF report image displays the default image
  **And** the PDF header displays the default text
  **And** the PDF footer displays the default text
  
  ***
  To manually test the configuration endpoint here are 2 scripts to update different configuration values.
  
[switch-toggle.zip](https://github.com/wazuh/wazuh-kibana-app/files/9735686/switch-toggle.zip)


To use it, first unzip the files in the same folder and allow both _executing the file as a program_. Afterwards you can run the `manual-test-switch-toggle.sh` script.

You can also use **wz-api-conf.sh** script with custom configuration parameters, like:
```
\- s setting_name
\- v setting_value
```

eg.
`./wz-api-conf.sh -s disabled_roles -v [\"disabled\"]`

#### Output

```

./manual-test-switch-toggle.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   398  100   398    0     0    418      0 --:--:-- --:--:-- --:--:--   417
{"customization.enabled":25000}
{"statusCode":400,"error":"Bad Request","message":"[request body.customization.enabled]: expected value of type [boolean] but got [number]"}  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   398  100   398    0     0    884      0 --:--:-- --:--:-- --:--:--   884
{"customization.enabled":true}
{"data":{"requiresRunningHealthCheck":false,"requiresReloadingBrowserTab":false,"requiresRestartingPluginPlatform":false,"updatedConfiguration":{"customization.enabled":true}}}  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   398  100   398    0     0    852      0 --:--:-- --:--:-- --:--:--   852
{"customization.enabled":false}
{"data":{"requiresRunningHealthCheck":false,"requiresReloadingBrowserTab":false,"requiresRestartingPluginPlatform":false,"updatedConfiguration":{"customization.enabled":false}}}

```
  
## Automatic tests
    
- Tests
  - `common/services/settings.test.ts`
  - `server/routes/wazuh-utils/wazuh-utils.test.ts`
  - `common/plugin-settings.test.ts`
  
```shell
yarn test:jest common/services/settings.test.ts server/routes/wazuh-utils/wazuh-utils.test.ts common/plugin-settings.test.ts
```
